src/Widget/Translation/Service/FilterForKeyWithFallback.php

```diff
@@ -21,7 +21,14 @@ public function __invoke(array $values, $preferredLanguage, $mainLanguage = 'nl'
         }
 
         if (!$this->hasPreferred($values, $preferredLanguage)) {
-            return isset($values[$this->fallBackLanguage]) ? $values[$this->fallBackLanguage] : $values[$mainLanguage];
+            if (isset($values[$this->fallBackLanguage])) {
```

Comment:
Can you this condition into a function like I did the line before, so you would have something like:
```
if ($this->hasFallbackLanguage($value)){
//
}
```
which, in my opinion, improves readability because as you read the code you can tell what this if the condition is checking without spending time on figuring out the logic of the condition itself.

It's technique I learned from Clean Code book and use it regularly.